tests/defragment.rs

```diff
@@ -0,0 +1,138 @@
+                },
+                Some(DataType::DoubleFloat) => {
+                    let mut data: Vec<f64> = vec![0.0; channel_length as usize];
+                    input_file.read_channel(&channel_path, &mut data).unwrap();
+                    writer.write_channels(&[channel_path], &data, tedium::DataLayout::Interleaved).unwrap();
+                },
+                Some(DataType::I8) => {
+                    let mut data: Vec<i8> = vec![0; channel_length as usize];
+                    input_file.read_channel(&channel_path, &mut data).unwrap();
+                    writer.write_channels(&[channel_path], &data, tedium::DataLayout::Interleaved).unwrap();
+                },
+                Some(DataType::I16) => {
+                    let mut data: Vec<i16> = vec![0; channel_length as usize];
+                    input_file.read_channel(&channel_path, &mut data).unwrap();
+                    writer.write_channels(&[channel_path], &data, tedium::DataLayout::Interleaved).unwrap();
+                },
+                Some(DataType::I32) => {
+                    let mut data: Vec<i32> = vec![0; channel_length as usize];
+                    input_file.read_channel(&channel_path, &mut data).unwrap();
+                    writer.write_channels(&[channel_path], &data, tedium::DataLayout::Interleaved).unwrap();
+                },
+                Some(DataType::I64) => {
+                    let mut data: Vec<i64> = vec![0; channel_length as usize];
+                    input_file.read_channel(&channel_path, &mut data).unwrap();
+                    writer.write_channels(&[channel_path], &data, tedium::DataLayout::Interleaved).unwrap();
+                },
+                Some(DataType::U8) => {
+                    let mut data: Vec<u8> = vec![0; channel_length as usize];
+                    input_file.read_channel(&channel_path, &mut data).unwrap();
+                    writer.write_channels(&[channel_path], &data, tedium::DataLayout::Interleaved).unwrap();
+                },
+                Some(DataType::U16) => {
+                    let mut data: Vec<u16> = vec![0; channel_length as usize];
+                },
+                Some(DataType::U32) => {
+                    let mut data: Vec<u32> = vec![0; channel_length as usize];
+                    input_file.read_channel(&channel_path, &mut data).unwrap();
+                    writer.write_channels(&[channel_path], &data, tedium::DataLayout::Interleaved).unwrap();
+                },
+                Some(DataType::U64) => {
+                    let mut data: Vec<u64> = vec![0; channel_length as usize];
+                    input_file.read_channel(&channel_path, &mut data).unwrap();
+                    writer.write_channels(&[channel_path], &data, tedium::DataLayout::Interleaved).unwrap();
+                },
+                Some(data_type) => println!("Unsupported data type: {}", data_type),
+                None => println!("None"),
+            }
+
```

Comment:
@JamesMc86 this whole block could be refactored into a `copy_channel_data` high-level API method